### PR TITLE
Fix false positive in root-no-standard-properties

### LINKED ISSUE #929: Rule root-no-standard-property throwing warning when :root is in a :not() statement i.e. :not(:root)
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Potential bug. Currently when parsing with `root-no-standard-properties`, it is reading rules such as:

```
svg:not(:root) {
  overflow: hidden;
}
```

as a `:root` rule. This is a style property found in `normalize.css`. Not sure if this is being correctly identified as a warning.

> Which rule, if any, is this issue related to?

root-no-standard-properties

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
   "root-no-standard-properties": true
  }
}
```

> Which version of stylelint are you using?

`5.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Gulp plugin with PostCSS

### PR DESCRIPTION
**Pull request by m-allanson**
Closes #929

### CODE REVIEW

File: src/rules/root-no-standard-properties/index.js [hunk 1]
```diff
@@ -19,6 +19,12 @@ export default function (actual) {
     root.walkRules(rule => {
       if (rule.selector.indexOf(":root") === -1) { return }
 
+      // ignore :not(:root) selectors
+      for (const selector of rule.selectors) {
+        const notSelectorWithRoot = selector.indexOf(":not") !== -1 && selector.indexOf(":root") > 0
```
Comment (davidtheclark) [inline src/rules/root-no-standard-properties/index.js]: These means of identification doesn't seem precise enough to me. Wouldn't this allow the following to pass, and shouldn't: `.foo, :root, .bar:not(:hover) {}`?

**Comment by m-allanson** [comment]
@davidtheclark That rule would have passed, but I agree that the selector identification should be more precise. 

I've attempted to fix this by using `postcss-selector-parser` (hopefully in an idiomatic manner).

**Comment by davidtheclark** [comment]
This looks nice to me. @jeddy3 do you anticipate any problems, or should we merge?

**Comment by jeddy3** [comment]
Looks super to me. Thanks @m-allanson !

**Pull request merged**